samcli/commands/sync/command.py

```diff
@@ -373,31 +374,49 @@ def do_cli(
                                 dependency_layer,
                             )
                         else:
-                            execute_infra_contexts(build_context, package_context, deploy_context)
+                            code_sync_resources = execute_infra_contexts(build_context, package_context, deploy_context)
+
+                            if code_sync_resources:
+                                resource_ids = [str(resource) for resource in code_sync_resources]
+
+                                LOG.info("The following resources will be code synced for an update: %s", resource_ids)
+                                execute_code_sync(
+                                    template_file,
+                                    build_context,
+                                    deploy_context,
+                                    sync_context,
+                                    resource_ids,  # type: ignore
+                                    None,
+                                    dependency_layer,
+                                )
 
 
 def execute_infra_contexts(
     build_context: "BuildContext",
     package_context: "PackageContext",
     deploy_context: "DeployContext",
-) -> None:
+) -> Optional[Set[ResourceIdentifier]]:
```

Comment:
I would recommend returning empty set rather than `Optional[Set]` to make implementation easier on the other side.

Reply:
Having None vs an empty set can be helpful to determine if it means the sync infra executed vs. no code sync updates are required.

Reply:
I see, but in that case, I would recommend returning explicit field to indicate that. Hiding such information behind being `None` or `non-None` would make it hard to maintain in the future.

Reply:
AFAIK this is a common behaviour in our code base to check empty list/dict vs. None for such three-way possible returns, for better readability a can return an infra sync response object here, which has fields indicating such differences